### ISSUE TITLE
IOs issues with MsWebsocket transport

### ISSUE THREAD

**Issue by marto83** [ably/ably-dotnet#151]
A few months ago we had an issue with the MsWebsocket transport and IOS and that's we use WebSocket4Net as the default implementation there. 

The issue: 

The connection will be made but the first time you try and send a message you get the following error: 

> Transport State: Error. Error is Cannot access a disposed object. Object name: 'System.Net.Sockets.NetworkStream'.. at System.Net.WebConnection.Read (System.Net.HttpWebRequest request, System.Byte[] buffer, System.Int32 offset, System.Int32 size) <0x1007f2e40 + 0x000b4> in <dfac2ad8ae7c42209218a908c3070036#9a571ba238c025561306ec1f66fd83e6>:0 at System.Net.WebSockets.ClientWebSocket+<>c__DisplayClass35_0.<ReceiveAsync>b__0 () <0x1007fade0 + 0x002d7> in <dfac2ad8ae7c42209218a908c3070036#9a571ba238c025561306ec1f66fd83e6>:0 at System.Threading.Tasks.Task`1[TResult].InnerInvoke () <0x1002f1550 + 0x00057> in <2f4074c3120b4d80802e10af84b67d41#9a571ba238c025561306ec1f66fd83e6>:0 at System.Threading.Tasks.Task.Execute () <0x1002f60b0 + 0x00043> in <2f4074c3120b4d80802e10af84b67d41#9a571ba238c025561306ec1f66fd83e6>:0 --- End of stack trace from previous location where exception was thrown --- at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x1003298f0 + 0x00028> in <2f4074c3120b4d80802e10af84b67d41#9a571ba238c025561306ec1f66fd83e6>:0 at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x10032c670 + 0x000d3> in <2f4074c3120b4d80802e10af84b67d41#9a571ba238c025561306ec1f66fd83e6>:0 at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x10032c5d0 + 0x0007f> in <2f4074c3120b4d80802e10af84b67d41#9a571ba238c025561306ec1f66fd83e6>:0 at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x10032c570 + 0x00047> in <2f4074c3120b4d80802e10af84b67d41#9a571ba238c025561306ec1f66fd83e6>:0 at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () <0x10032cd80 + 0x0001b> in <2f4074c3120b4d80802e10af84b67d41#9a571ba238c025561306ec1f66fd83e6>:0 at IO.Ably.Transport.MsWebSocketConnection+<Receive>d__22.MoveNext () <0x1015ba960 + 0x00333> in <c4e321e6fb8e41ab8e5d206c4e805416#9a571ba238c025561306ec1f66fd83e6>:0

It happens here https://github.com/ably/ably-dotnet/blob/dotnet-standard/src/IO.Ably.Shared/Transport/MsWebSocketConnection.cs#L135

There is an issue open with Microsoft with #117071116022574 but I didn't have enough time to create a sample application where we were experiencing this happening. They have asked for a sample only including the code from our Ms Transport implementation that shows off the error. If you manage to recreate the error, Matt will need to email Ms support with the sample so they can reopen the case and investigate it.

The first step will be to switch and test the MsTransport layer on an iOS application and confirm it's still happening with the latest version of Xamarin.

**Comment by withakay** [comment]
I have setup a [test app](https://github.com/withakay/app-monkeychat/tree/ms-websocket-test) using the latest Xamarin release and cannot repeat this issue in the iOS simulator.

**Comment by withakay** [comment]
I have looked into this and discussed with @marto83. When the issue was originally reported that was based on an app built against Xamarin.iOS 10.4, at the time of writing the current version is Xamarin.iOS 11.3. Additionally Microsoft also could also not repeat the bug, they were using Xamarin.iOS 10.10, incidentally 10.10 was the version that moved to using mono 5.0 (11.3 uses mono 5.4, 10.4 was using mono 4.8), the significance of which is that 5.0 brought some code over from the newly open sourced .Net.

I have tried to setup an older version of Xamarin.iOS in order to recreate this bug but have been unable to get it to work on my system, it appears to require an older version of Xcode, which I can't install on macOS High Sierra. 

If we can be confident this issue is no longer a problem then this would provide a neat solution for https://github.com/ably/ably-dotnet/issues/161, as we could simply remove the websocket4net dependency.

WDYT @mattheworiordan ?
